org.eclipse.dawnsci.analysis.dataset/src/org/eclipse/dawnsci/analysis/dataset/MockDynamicLazyDataset.java

```diff
@@ -0,0 +1,132 @@
+package org.eclipse.dawnsci.analysis.dataset;
+
+import java.io.IOException;
+
+import org.eclipse.january.IMonitor;
+import org.eclipse.january.dataset.Dataset;
+import org.eclipse.january.dataset.IDataListener;
+import org.eclipse.january.dataset.IDataset;
+import org.eclipse.january.dataset.IDatasetChangeChecker;
+import org.eclipse.january.dataset.IDynamicDataset;
+import org.eclipse.january.dataset.LazyDynamicDataset;
+import org.eclipse.january.dataset.ShapeUtils;
+import org.eclipse.january.dataset.SliceND;
+import org.eclipse.january.io.ILazyLoader;
+
+public class MockDynamicLazyDataset extends LazyDynamicDataset {
+
+	private static final long serialVersionUID = 1L;
+	private int[][] shapeArrays;
+	private int count = 1;
+	private Dataset parent;
+	public boolean allowIncrement = false;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make allowIncrement a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)